docs/topics/addons.rst

```diff
@@ -0,0 +1,348 @@
+.. _topics-addons:
+
+=======
+Add-ons
+=======
+
+Scrapy's add-on system is a framework which unifies managing and configuring
+components that extend Scrapy's core functionality, such as middlewares,
+extensions, or pipelines. It provides users with a plug-and-play experience in
+Scrapy extension management, and grants extensive configuration control to
+developers.
+
+
+Activating and configuring add-ons
+==================================
+
+Add-ons and their configuration live in Scrapy's
+:class:`~scrapy.addons.AddonManager`. During Scrapy's start-up process, and
+only then, the add-on manager will read a list of enabled add-ons and their
+configurations from your ``ADDONS`` setting.
+
+The ``ADDONS`` setting is a dict in which every key is an addon class or its
+import path and the vaoue is its priority.
+
+The configuration of an add-on, if necessary at all, is stored as a dictionary
+setting whose name is the uppercase add-on name.
+
+This is an example where two add-ons (in this case with one requiring no
+configuration) are enabled/configured in a project's ``settings.py``::
+
+    ADDONS = {
+        'path.to.someaddon': 0,
+        path.to.someaddon2: 1,
+    }
+
+    SOMEADDON = {
+        'some_config': True,
+    }
+
+Enabling and configuring add-ons within Python code
+---------------------------------------------------
+
+The :class:`~scrapy.addons.AddonManager` will only read from Scrapy's settings
+*at the beginning* of Scrapy's start-up process.
+Afterwards, i.e. as soon as the :class:`~scrapy.addons.AddonManager` is
+populated, changing the ``ADDONS`` setting or any of the add-on
+configuration dictionary settings will have no effect.
+
+If you want to enable, disable, or configure add-ons in Python code, for example
+when writing your own add-on, you will have to use the
+:class:`~scrapy.addons.AddonManager`. You can access the add-on manager through
+either ``crawler.addons`` or, if you are writing an add-on, through the
+``addons`` argument of the :meth:`update_addons` callback. The add-on manager
+provides many useful methods and attributes to facilitate interacting with the
+add-ons framework, e.g.:
```

Comment:
I think we can drop this feature, and consider adding it if there is a real use case.